57.md

```diff
@@ -0,0 +1,146 @@
+NIP-57
+======
+
+Lightning Zaps
+--------------
+
+`draft` `optional` `author:jb55` `author:kieran`
+
+This NIP defines a new note type called a lightning zap of kind `9735`. These represent paid lightning invoice receipts sent by a lightning node called the `zapper`. We also define another note type of kind `9734` which are `zap request` notes, which will be described in this document.
+
+Having lightning receipts on nostr allows clients to display lightning payments from entities on the network. These can be used for fun or for spam deterrence.
+
+
+## Definitions
+
+`zapper` - the lightning node or service that sends zap notes (kind `9735`)
+
+`zap request` - a note of kind `9734` created by the person zapping
+
+`zap invoice` - the bolt11 invoice fetched from a custom lnurl endpoint which contains a `zap request` note
+
+
+## Protocol flow
+
+### Client side
+
+1. Calculate the lnurl pay request url for a user from the lud06 or lud16 field on their profile
+
+2. Fetch the lnurl pay request static endpoint (`https://host.com/.well-known/lnurlp/user`) and gather the `allowsNostr` and `nostrPubkey` fields. If `allowsNostr` exists and it is `true`, and if `nostrPubkey` exists and is true, associate this information with the user. The `nostrPubkey` is the `zapper`'s pubkey, and it is used to authorize zaps sent to that user.
+
```

Comment:
What is the point of `allowsNostr`, what difference does it server vs just having `nostrPubkey` defined or not

Reply:
https://github.com/nostr-protocol/nips/pull/224#issuecomment-1418256137

Reply:
also:

> `nostrPubkey` exists and is true

should probably be ``nostrPubkey` exists and is a valid BIP 340 public key

Reply:
thanks!